__init__.py

```diff
@@ -33,15 +33,55 @@
 __version__ = '1.0'
 __author__  = 'Forte Developers'
 
+import sys
+
 # Load Python modules
 from .pymodule import *
 
+from .register_forte_options import *
+
 # Load C++ plugin
-#import os
-import psi4
 from .forte import *
 
-# Register options with psi
-options = psi4.core.get_options()
-options.set_current_module('FORTE')
-forte.read_options(psi4.core.get_options())
+# Create a ForteOptions object (stores all options)
+forte_options = forte.ForteOptions();
+
+# Register options defined in Forte in the forte_options object
+register_forte_options(forte_options) # py-side
+forte.read_options(forte_options) # c++-side
+
+# If we are running psi4, push the options defined in forte_options to psi
+if 'psi4' in sys.modules:
+    psi_options = psi4.core.get_options()
+    psi_options.set_current_module('FORTE')
+    forte_options.push_options_to_psi4(psi_options)
+
```

Comment:
I think ForteOptions does have array functionality for int and double

Reply:
Thanks. Implemented.